backend-shared/config/db/repo_cred_test.go

```diff
@@ -0,0 +1,145 @@
+package db_test
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/managed-gitops/backend-shared/config/db"
+)
+
+var _ = Describe("RepositoryCredentials Tests", func() {
+	var (
+		err                  error
+		ctx                  context.Context
+		clusterUser          *db.ClusterUser
+		clusterCredentials   db.ClusterCredentials
+		gitopsEngineCluster  db.GitopsEngineCluster
+		gitopsEngineInstance db.GitopsEngineInstance
+		dbq                  db.AllDatabaseQueries
+	)
+
+	When("When ClusterUser, ClusterCredentials, GitopsEngine and GitopsInstance exist", func() {
+		BeforeEach(func() {
+			// Connect to the database (the connection closes at AfterEach)
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			dbq, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+
+			// Satisfy the foreign key constraint 'fk_clusteruser_id'
+			// aka: ClusterUser.Clusteruser_id) required for 'repo_cred_user_id'
+			clusterUser = &db.ClusterUser{
+				Clusteruser_id: "test-repocred-user-id",
+				User_name:      "test-repocred-user",
+			}
+			err = dbq.CreateClusterUser(ctx, clusterUser)
+			Expect(err).To(BeNil())
+
+			// Satisfy the foreign key constraint 'fk_gitopsengineinstance_id'
+			// aka: GitOpsEngineInstance.Gitopsengineinstance_id) required for 'repo_cred_gitopsengineinstance_id'
+			clusterCredentials = db.ClusterCredentials{
+				Clustercredentials_cred_id:  "test-repocred-Clustercredentials_cred_id",
+				Host:                        "test-repocred-host",
+				Kube_config:                 "test-repocred-kube-config",
+				Kube_config_context:         "test-repocred-kube-config-context",
+				Serviceaccount_bearer_token: "test-repocred-serviceaccount_bearer_token",
+				Serviceaccount_ns:           "test-repocred-Serviceaccount_ns",
+			}
+
+			err = dbq.CreateClusterCredentials(ctx, &clusterCredentials)
+			Expect(err).To(BeNil())
+
+			gitopsEngineCluster = db.GitopsEngineCluster{
+				Gitopsenginecluster_id: "test-repocred-Gitopsenginecluster_id",
+				Clustercredentials_id:  clusterCredentials.Clustercredentials_cred_id,
+			}
+
+			err = dbq.CreateGitopsEngineCluster(ctx, &gitopsEngineCluster)
+			Expect(err).To(BeNil())
+
+			gitopsEngineInstance = db.GitopsEngineInstance{
+				Gitopsengineinstance_id: "test-repocred-Gitopsengineinstance_id",
+				Namespace_name:          "test-repocred-Namespace_name",
+				Namespace_uid:           "test-repocred-Namespace_uid",
+				EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+
+			err = dbq.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			// Delete Cluster User, Cluster Credentials, Gitops Engine Cluster, Gitops Engine Instance.
+			_, err = dbq.DeleteGitopsEngineInstanceById(ctx, gitopsEngineInstance.Gitopsengineinstance_id)
```

Comment:
I actually think I don't need this section at all, since I am calling `db.SetupForTestingDBGinkgo()` in the beginning, and other tests do the same. I guess I did that back then, because I did not have yet implemented the code to delete repo_cred resources as part of the `db.SetupForTestingDBGinkgo()` function.